tests/task_t_1.spec.js

```diff
@@ -48,3 +44,49 @@ test('T4: Check that URL shortening feature works', async ({ page }) => {
   const newPageURL = await page.url();
   expect(newPageURL).toBe("https://www.wikipedia.org/");
 });
+
+test('T5: Check that intitle functionality works', async ({ page }) => {
+  const searchInputField = page.locator('#search_form_input_homepage');
+  await searchInputField.fill('intitle:panda');
+  await page.click('#search_button_homepage');
+
+  const results = await page.evaluate(() => Array.from(document.querySelectorAll('div [data-nir="1"]'), element => element.textContent));
+        results.forEach(result => {
+            expect(result.toLowerCase()).toContain("panda");
+        });
+});
+
+test('T6: Check that automatic navigation to first result works', async ({ page }) => {
+  const searchInputField = page.locator('#search_form_input_homepage');
+  await searchInputField.fill('!w Lithuania');
+  await page.click('#search_button_homepage');
+  const newLoadedPage = await page.url();
+  await page.locator('h1#firstHeading:has-text("Lithuania")').toBeVisible;
+});
+
+
+const passwordLengthsPositive = ['8', '16', '64']
+passwordLengthsPositive.forEach(passwordLength => {
+  test(`T7: Check that generated passwords for length ${passwordLength} are of correct length`, async ({ page }) => {
+    const searchInputField = page.locator('#search_form_input_homepage');
+    await searchInputField.fill("password " + passwordLength);
+    await page.click('#search_button_homepage');
+
+    const generatedPassword = await page.textContent('h3.c-base__title');
+    expect(generatedPassword.length).toEqual(+passwordLength);
+  });
+});
+
+
+const passwordLengthsNegative = ['7', '65']
+passwordLengthsNegative.forEach(passwordLength => {
+  test(`T8: Check that passwords are not generated for lengths ${passwordLength}`, async ({ page }) => {
+    const searchInputField = page.locator('#search_form_input_homepage');
+    await searchInputField.fill("password " + passwordLength);
+    await page.click('#search_button_homepage');
+
+    const generatedPassword = await page.locator('h3.c-base__title >> visible=false');
```

Comment:
I would suggest not using attributes in the selector description, but instead do something like 

 `const isPasswordContentVisible = await page.isVisible(".c-base__title");
    expect(isPasswordContentVisible).toBe(false)`

Reply:
@jevgenijusmarinuskinas  thank you so much for taking time and all the suggestions 😎